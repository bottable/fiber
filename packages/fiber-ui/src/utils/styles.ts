import {
  compose,
  space,
  color,
  typography,
  height,
  SpaceProps,
  ColorProps,
  TypographyProps
} from 'styled-system'

export const styleComposition = compose(space, height, color, typography)

export type StyleProps = SpaceProps & ColorProps & TypographyProps
