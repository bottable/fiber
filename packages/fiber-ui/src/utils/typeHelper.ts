export function withProperties<A, B>(component: A, properties: B): A & B {
  Object.keys(properties).forEach((key) => {
    // @ts-ignore
    component[key] = properties[key]
  })
  return component as A & B
}
