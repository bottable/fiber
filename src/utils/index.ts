export * from './typeHelper'
export * from './mergeProps'
