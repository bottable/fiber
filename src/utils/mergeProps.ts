import React from 'react'

export type MergeElementProps<
  T extends React.ElementType,
  P extends object = {}
> = Omit<React.ComponentPropsWithRef<T>, keyof P> & P
