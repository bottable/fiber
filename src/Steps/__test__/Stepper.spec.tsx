import { Steps } from '..'

import React from 'react'
import { render } from 'test/utils'
import PersonIcon from '@material-ui/icons/Person'
import VerifiedUserIcon from '@material-ui/icons/VerifiedUser'
import PaymentIcon from '@material-ui/icons/Payment'
import InsertEmoticonIcon from '@material-ui/icons/InsertEmoticon'

describe('Steps', () => {
  it('renders correctly', () => {
    const steps = render(
      <>
        <Steps current={1}>
          <Steps.Step title='first' subtitle='subtitle' />
          <Steps.Step title='second' subtitle='subtitle' />
          <Steps.Step title='third' subtitle='subtitle' />
        </Steps>
        <Steps current={1} vertical>
          <Steps.Step title='first' subtitle='subtitle'>
            <div
              style={{
                width: 200,
                border: '1px solid black',
                padding: 25
              }}
            >
              First Content
            </div>
          </Steps.Step>
          <Steps.Step title='second' subtitle='subtitle'>
            <div
              style={{
                width: 200,
                border: '1px solid black',
                padding: 25
              }}
            >
              Second Content
            </div>
          </Steps.Step>
          <Steps.Step title='third' subtitle='subtitle'>
            <div
              style={{
                width: 200,
                border: '1px solid black',
                padding: 25
              }}
            >
              Third Content
            </div>
          </Steps.Step>
        </Steps>
        <Steps current={1}>
          <Steps.Step title='Login' icon={<PersonIcon />} />
          <Steps.Step title='Verification' icon={<VerifiedUserIcon />} />
          <Steps.Step title='Pay' icon={<PaymentIcon />} />
          <Steps.Step title='Done' icon={<InsertEmoticonIcon />} />
        </Steps>
      </>
    )

    expect(steps).toMatchSnapshot()
  })
})
