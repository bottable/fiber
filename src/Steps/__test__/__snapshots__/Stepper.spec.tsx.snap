// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps renders correctly 1`] = `
Array [
  <div
    className="sc-AxiKw iVZTBv"
  >
    <div
      className="sc-AxhCb jxrXPr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxhUy hLyCPb"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 5H1v16c0 1.1.9 2 2 2h16v-2H3V5zm11 10h2V5h-4v2h2v8zm7-14H7c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V3c0-1.1-.9-2-2-2zm0 16H7V3h14v14z"
          />
        </svg>
      </div>
      <div
        className="sc-AxgMl ctSGnM"
      >
        <div
          className="sc-AxheI bqoywY"
        >
          first
        </div>
        <div
          className="sc-Axmtr eaFFCX"
        >
          subtitle
        </div>
      </div>
    </div>
    <div
      className="sc-AxhCb jxrXPr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxhUy crOSLQ"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 5H1v16c0 1.1.9 2 2 2h16v-2H3V5zm18-4H7c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V3c0-1.1-.9-2-2-2zm0 16H7V3h14v14zm-4-4h-4v-2h2c1.1 0 2-.89 2-2V7c0-1.11-.9-2-2-2h-4v2h4v2h-2c-1.1 0-2 .89-2 2v4h6v-2z"
          />
        </svg>
      </div>
      <div
        className="sc-AxgMl ctSGnM"
      >
        <div
          className="sc-AxheI bqoywY"
        >
          second
        </div>
        <div
          className="sc-Axmtr eaFFCX"
        >
          subtitle
        </div>
      </div>
    </div>
    <div
      className="sc-AxhCb jxrXPr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxhUy crOSLQ"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 1H7c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V3c0-1.1-.9-2-2-2zm0 16H7V3h14v14zM3 5H1v16c0 1.1.9 2 2 2h16v-2H3V5zm14 8v-1.5c0-.83-.67-1.5-1.5-1.5.83 0 1.5-.67 1.5-1.5V7c0-1.11-.9-2-2-2h-4v2h4v2h-2v2h2v2h-4v2h4c1.1 0 2-.89 2-2z"
          />
        </svg>
      </div>
      <div
        className="sc-AxgMl ctSGnM"
      >
        <div
          className="sc-AxheI dQbFFX"
        >
          third
        </div>
        <div
          className="sc-Axmtr eaFFCX"
        >
          subtitle
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-AxiKw BkSBb"
  >
    <div
      className="sc-AxhCb eCvjWQ"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxhUy hauDQr"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 5H1v16c0 1.1.9 2 2 2h16v-2H3V5zm11 10h2V5h-4v2h2v8zm7-14H7c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V3c0-1.1-.9-2-2-2zm0 16H7V3h14v14z"
          />
        </svg>
      </div>
      <div
        className="sc-AxgMl ctSGnM"
      >
        <div
          className="sc-AxheI dQbFFX"
        >
          first
        </div>
        <div
          className="sc-Axmtr gNnseu"
        >
          subtitle
        </div>
        <div
          className="sc-AxirZ kiMEmg"
        >
          <div
            style={
              Object {
                "border": "1px solid black",
                "padding": 25,
                "width": 200,
              }
            }
          >
            First Content
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-AxhCb eCvjWQ"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxhUy crOSLQ"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 5H1v16c0 1.1.9 2 2 2h16v-2H3V5zm18-4H7c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V3c0-1.1-.9-2-2-2zm0 16H7V3h14v14zm-4-4h-4v-2h2c1.1 0 2-.89 2-2V7c0-1.11-.9-2-2-2h-4v2h4v2h-2c-1.1 0-2 .89-2 2v4h6v-2z"
          />
        </svg>
      </div>
      <div
        className="sc-AxgMl ctSGnM"
      >
        <div
          className="sc-AxheI dQbFFX"
        >
          second
        </div>
        <div
          className="sc-Axmtr gNnseu"
        >
          subtitle
        </div>
        <div
          className="sc-AxirZ kiMEmg"
        >
          <div
            style={
              Object {
                "border": "1px solid black",
                "padding": 25,
                "width": 200,
              }
            }
          >
            Second Content
          </div>
        </div>
      </div>
    </div>
    <div
      className="sc-AxhCb eCvjWQ"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxhUy crOSLQ"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 1H7c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V3c0-1.1-.9-2-2-2zm0 16H7V3h14v14zM3 5H1v16c0 1.1.9 2 2 2h16v-2H3V5zm14 8v-1.5c0-.83-.67-1.5-1.5-1.5.83 0 1.5-.67 1.5-1.5V7c0-1.11-.9-2-2-2h-4v2h4v2h-2v2h2v2h-4v2h4c1.1 0 2-.89 2-2z"
          />
        </svg>
      </div>
      <div
        className="sc-AxgMl ctSGnM"
      >
        <div
          className="sc-AxheI dQbFFX"
        >
          third
        </div>
        <div
          className="sc-Axmtr gNnseu"
        >
          subtitle
        </div>
        <div
          className="sc-AxirZ kiMEmg"
        >
          <div
            style={
              Object {
                "border": "1px solid black",
                "padding": 25,
                "width": 200,
              }
            }
          >
            Third Content
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="sc-AxiKw iVZTBv"
  >
    <div
      className="sc-AxhCb jxrXPr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxhUy hLyCPb"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
      <div
        className="sc-AxgMl ctSGnM"
      >
        <div
          className="sc-AxheI bqoywY"
        >
          Login
        </div>
      </div>
    </div>
    <div
      className="sc-AxhCb jxrXPr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxhUy crOSLQ"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm-2 16l-4-4 1.41-1.41L10 14.17l6.59-6.59L18 9l-8 8z"
          />
        </svg>
      </div>
      <div
        className="sc-AxgMl ctSGnM"
      >
        <div
          className="sc-AxheI bqoywY"
        >
          Verification
        </div>
      </div>
    </div>
    <div
      className="sc-AxhCb jxrXPr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxhUy crOSLQ"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V6c0-1.11-.89-2-2-2zm0 14H4v-6h16v6zm0-10H4V6h16v2z"
          />
        </svg>
      </div>
      <div
        className="sc-AxgMl ctSGnM"
      >
        <div
          className="sc-AxheI bqoywY"
        >
          Pay
        </div>
      </div>
    </div>
    <div
      className="sc-AxhCb jxrXPr"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="sc-AxhUy crOSLQ"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm3.5-9c.83 0 1.5-.67 1.5-1.5S16.33 8 15.5 8 14 8.67 14 9.5s.67 1.5 1.5 1.5zm-7 0c.83 0 1.5-.67 1.5-1.5S9.33 8 8.5 8 7 8.67 7 9.5 7.67 11 8.5 11zm3.5 6.5c2.33 0 4.31-1.46 5.11-3.5H6.89c.8 2.04 2.78 3.5 5.11 3.5z"
          />
        </svg>
      </div>
      <div
        className="sc-AxgMl ctSGnM"
      >
        <div
          className="sc-AxheI dQbFFX"
        >
          Done
        </div>
      </div>
    </div>
  </div>,
]
`;
